### PR TITLE
feat: Read default dev port from .env

### PR DESCRIPTION
**Pull request by pke**
Adds support for specifying the default port via environment variable `PORT`.

`dotenv` is used to read that variable from `.env` files.
The default port remains 3000.

Helps to solve #2613

What tests are required, where do I hook them in at best?

### CODE REVIEW

**Review by styfle — CHANGES_REQUESTED** [review]
This will cause bugs for frameworks that launch their own dev server such as Gatsby.

Currently, `now dev` uses the `PORT` environment variable to proxy requests to the framework's dev server.

https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/gatsby-cli.md#develop

**Comment by pke** [comment]
Thanks for your feedback.

> Currently, `now dev` uses the `PORT` environment variable to proxy requests to the framework's dev server.

I am not sure I understand what the implications are here. The code I changed is run when framework dev servers are run or not? Why does it matter, when `now` not mounts the port itself, reading the env var does no harm, does it?

**Review by TooTallNate — COMMENTED** [review]
The code LGTM, but would it be possible to add a new test case for this behavior?

**Review by styfle — COMMENTED** [review]
On second thought, the PORT is assigned to the child process so this PR wouldn't be an issue

https://github.com/zeit/now/blob/master/packages/now-static-build/src/index.ts#L351

Disregard my previous comment.

**Comment by pke** [comment]
Do you have any idea where such test could live? Do we already have tests that involve the `.env` file?

**Comment by TooTallNate** [comment]
I don't think it needs to read the `.env` file specifically, but test that spawning the process with the `PORT` env var defined would be enough in my opinion.

**Comment by pke** [comment]
> I don't think it needs to read the `.env` file specifically, but test that spawning the process with the `PORT` env var defined would be enough in my opinion.

What if it does not spawn a process but provides the server itself? None of the mentioned scenarios collides with the marginal change this PR provides.
I like to see `now dev` working with standard node apps first and with other frameworks as an addon (that's why it requires an entry in the `now.json` file to work with other frameworks).

**Comment by andymckay** [comment]
Based on the fact that this has been open for over 2 years, I'm going to close it in an effort to clean up the backlog. If you want to keep working on it, my apologies and please re-open it.

File: packages/now-cli/src/commands/dev/dev.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import path from 'path';
+import 'dotenv/config';
```
Comment (jerrygreen) [inline packages/now-cli/src/commands/dev/dev.ts:2]: Shouldn’t this line break users projects that aren’t having `dotenv`? I mean, this package isn’t listed in dependencies for the cli, it only listed as `devDependency`. If we fix that, I would say LGTM.

**Review by jerrygreen — COMMENTED** [review]
Yes but why this wasn’t been merged in the first place? Because it didn’t had a test?

It’s a good feature: having to be able to read CLI parameters from env as an option is a good feature overall (well unless they don’t override cli parameters, in which case they don’t).

`PORT` is one of the most important one, though I personally wouldn’t mind having env var equivalents for all cli parameters, so they can be stored in a file rather than passed to cli each time.

I only not sure why dotenv wasn’t been moved to package’s main dependencies.

**Pull request closed without merging**